presto-testing/src/main/java/io/prestosql/testing/CountingMockConnector.java

```diff
@@ -167,6 +202,103 @@ public String toString()
                     .add("listSchemasCount", listSchemasCount)
                     .add("listTablesCount", listTablesCount)
                     .add("getColumnsCount", getColumnsCount)
+                    .add("roleGrantCount", roleGrantCount)
+                    .toString();
+        }
+    }
+
+    public static class RoleGrantCounter
+    {
+        private final AtomicLong listRowGrantsCallsCounter = new AtomicLong();
+        private final AtomicLong rolesPushedCounter = new AtomicLong();
+        private final AtomicLong granteesPushedCounter = new AtomicLong();
+        private final AtomicLong limitPushedCounter = new AtomicLong();
```

Comment:
I would prefer to have bean to be immutable. `incrementListRoleGrants` could return new instance. Then you don't need two classes.

Reply:
The AtomicLongs for the other counters are mutable. I think this counter would be inherently mutable.
How do you handle atomicity between calls? Have the counter bean be an atomic reference and you update the reference with each new instance?

We're going to need 2 beans, right? One for counting and one to compare against during the test.
The more I think about it, the more I think the previous code was better.